trio/tests/test_highlevel_open_unix_stream.py

```diff
@@ -0,0 +1,35 @@
+import socket
+import tempfile
+
+import pytest
+
+from trio import open_unix_socket, Path
+
+try:
+    from socket import AF_UNIX
+except ImportError:
+    pytestmark = pytest.mark.skip("Needs unix socket support")
+
+
+async def test_open_unix_socket():
+    name = Path() / tempfile.gettempdir() / "test.sock"
+    try:
+        await name.unlink()
+    except OSError:
+        pass
+
+    # server socket
+    serv_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    serv_sock.bind(name.__fspath__())
+    serv_sock.listen(1)
+
+    # for some reason the server socket can accept the connection AFTER it has
+    # been opened, so use that to test here
```

Comment:
Yep, that's the point of the `listen(1)` :-). It means "allow at least 1 connection to accepted, and hold onto it in the kernel, before I call `accept`".

Reply:
Ah, I didn't know that. Been a *long* time since I've used raw sockets.